### ISSUE TITLE
Graphic: add  placeholder in case of missing image 

### ISSUE THREAD

**Issue by seloumi** [latex3/latex2e#178]
## Brief outline of the bug
Hi with `context`  when inserting images in cas of  images are not found a box is printed with label on it  showing that the requested image can't be  found, I think it's a good Idea  so my hope is to reproduce this feature on latex (if it is possible), thanks in advance.
I  know  there is `demo`  option which draw black rectangle in place of all included images, what I mean is  replacing only non existing images with box with label on it. 

## Minimal example showing the bug

```latex
\RequirePackage{latexbug}      
\documentclass{article}
\usepackage{graphicx}
\begin{document}
\begin{center}
\includegraphics[width=4cm]{myimage}
\end{center}
\end{document}
```

##   Image produced with an equivalent  code with context

![image](https://user-images.githubusercontent.com/8249073/64920893-d9d5c600-d7bc-11e9-83bc-d8fa721c46fa.png)

**Comment by dbitouze** [comment]
It would be a pity if this issue would be closed, so here is an activity :)

**Comment by davidcarlisle** [comment]
@davidcarlisle  yes we are trailing  the bot which meant it flagged a lot of issues last night. Actually we have configured it _not_ to auto close (just to flag unhandled issues)  but it currently has the default message..

**Comment by FrankMittelbach** [comment]
@davidcarlisle in case you have another cycle, else move it to spring

**Comment by josephwright** [comment]
I've been giving this some thought. Before we can tackle any code change, there's a conceptual issue. In ConTeXt, a missing graphic doesn't lead to a hard error: instead as observed a placeholder is used. In contrast, in LaTeX, a missing graphic leads to an error, so the user is expected to fix it before even looking at the output. Neither behaviour is 'more correct' than the other, but it means that adding a placeholder in LaTeX feels a bit odd to me.

**Comment by josephwright** [comment]
@davidcarlisle  ^^^

**Comment by davidcarlisle** [comment]
Note  `graphics` has always had code to show a boxed filename, but it only triggers it if the extension is used.

```
\documentclass{article}
\usepackage{graphicx}
\begin{document}
\begin{center}
%AA\includegraphics[width=4cm]{myimage}BB

AA\includegraphics[width=4cm]{myimage.png}BB


AA\includegraphics[width=4cm]{example-image}BB
\end{center}
\end{document}
```

produces a missing file error but then a display as requested:

<img width="285" height="302" alt="Image" src="https://github.com/user-attachments/assets/742102ac-b9dd-4930-8135-6446f5b8a060" />

It was a long time ago, but not doing this in the mssing extension case was a concious decision. We/I thought that as the file not existing with several potential extensions had already been used in the loop finding an extension, it was better to give a harder error.

It wouldn't be impossibe to re-visit this and show the boxed filename if no extension was given.

Separately LaTeX gives an error if it can not read a file to find a natural size (or a natural size is not given in the arguments)
and just gives a warning if a natural size is found but the image file is not. partly this is historical in that latex/dvips can not read the image file during the latex processing anyway so it does not have to exist until the dvips run.

This could be made a warning either always, or via a package option, although in general missing files in latex are handled as errors.

I (think?) I would now recommend keeping this as an error but if you scroll past the error it shows the boxed filename in the typeset uoutput in the "no extension" case.



If you ammend

```
    \ifx\Gin@ext\relax
         \@latex@error{File `#1' not found}%
         {I could not locate the file with any of these extensions:^^J%
          \Gin@extensions^^J\@ehc}%
    \else
```

to

```
\ifx\Gin@ext\relax
         \@latex@error{File `#1' not found}%
         {I could not locate the file with any of these extensions:^^J%
           \Gin@extensions^^J\@ehc}%
         \Gin@nat@height=1in %
         \Gin@nat@width =1in %
               \def\Gin@llx{0}%
      \let\Gin@lly\Gin@llx
      \def\Gin@urx{72}%
      \let\Gin@ury\Gin@urx
         \Gin@bboxtrue
         \def\Gread@{}%
         \Gin@setfile{}{}{#1}%
    \else
```


Then

```
\documentclass{article}
\usepackage{graphicx}
\begin{document}
\begin{center}
AA\includegraphics[width=4cm]{myimage}BB

AA\includegraphics[width=4cm]{myimage.png}BB


AA\includegraphics[width=4cm]{example-image}BB
\end{center}
\end{document}
```

produces

<img width="399" height="492" alt="Image" src="https://github.com/user-attachments/assets/089fb99c-783c-4f81-b285-77e287ffc926" />

if you scroll past two missing file errors.

**Comment by josephwright** [comment]
@davidcarlisle  Should i do a PR or will you?
